torch/csrc/autograd/python_saved_variable_hooks.cpp

```diff
@@ -45,39 +46,37 @@ namespace torch { namespace autograd {
     }
   }
 
-  std::mutex PyDefaultSavedVariableHooks::mutex_;
-  PyObject* PyDefaultSavedVariableHooks::pack_hook_(nullptr);
-  PyObject* PyDefaultSavedVariableHooks::unpack_hook_(nullptr);
-
   void PyDefaultSavedVariableHooks::set_hooks(py::function &pack_hook, py::function &unpack_hook) {
-    std::lock_guard<std::mutex> lock(mutex_);
+    PyObject *pack_hook_(nullptr), *unpack_hook_(nullptr);
```

Comment:
Do you need to initialize these to nullptr? You actually set them below.

Reply:
If I don't do it here as well, I get a clang-tidy error. So another option would be to add a no-lint comment